src/TreeSupport.cpp

```diff
@@ -22,10 +22,11 @@
 namespace cura
 {
 
-TreeSupport::TreeSupport(const SliceDataStorage& storage)
+namespace {
```

Comment:
The reasons you mention for making it anonymous namespace sound to me like perfect reasons to make this function a private class function.

By making it a static function in anonymous namespace you are bypassing the documentation rules.

Please make it into a private (static) function and document it in the header file.

Reply:
Done